iceoryx_utils/test/moduletests/test_posix_thread.cpp

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2020 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_utils/posix_wrapper/thread.hpp"
+#include "test.hpp"
+
+#include <atomic>
+#include <thread>
+
+using namespace ::testing;
+using namespace iox::posix;
+using namespace iox::cxx;
+
+class Thread_test : public Test
+{
+  public:
+    Thread_test()
+    {
+    }
+
+    void SetUp()
+    {
+        m_run = true;
+        m_thread = new std::thread(&Thread_test::threadFunc, this);
```

Comment:
Please use `pthread_create` instead of `std::thread~

Reply:
Ok understood, `native_handle_type` is not portable.

Reply:
Ok, didn't manage to port to the native `pthread_create` interface in less than 1h. It compiles but getting a segfault, when setting the name. Please do this in the thread wrapper story.